wpilibj/src/test/java/edu/wpi/first/wpilibj/UnitTestUtility.java

```diff
@@ -0,0 +1,91 @@
+package edu.wpi.first.wpilibj;
+
+import com.google.common.base.Stopwatch;
+
+import java.util.concurrent.TimeUnit;
+
+import edu.wpi.first.wpilibj.util.BaseSystemNotInitializedException;
+
+/**
+ * Utility class for configuring unit tests.
+ */
+public final class UnitTestUtility {
+  private UnitTestUtility() {
+        /* no-op */
+  }
+
+  /**
+   * Sets up the base system WPILib so that it does not rely on hardware.
+   */
+  public static void setupMockBase() {
+    try {
+      // Check to see if this has been setup
+      Timer.getFPGATimestamp();
+    } catch (BaseSystemNotInitializedException ex) {
+      // If it hasn't been then do this setup
+
+      HLUsageReporting.SetImplementation(new HLUsageReporting.Null());
+      RobotState.SetImplementation(new MockRobotStateInterface());
+      Timer.SetImplementation(new Timer.StaticInterface() {
+
+        @Override
+        public double getFPGATimestamp() {
+          return System.currentTimeMillis() / 1000.0;
+        }
+
+        @Override
+        public double getMatchTime() {
+          return 0;
+        }
+
+        @Override
+        public void delay(double seconds) {
+          try {
+            Thread.sleep((long) (seconds * 1e3));
+          } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Thread was interrupted", ex);
```

Comment:
I'm not very familiar with InterruptedException, but you interrupt the current thread and then throw a different exception?  Does it ever make it to the throw?